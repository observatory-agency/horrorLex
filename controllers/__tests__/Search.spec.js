const SearchController = require('../Search');

describe('SearchController', () => {
  let search;
  beforeEach(() => {
    search = new SearchController();
  });

  describe('constructor', () => {
    it('should set its templates', () => {
      expect(search.template).toMatchSnapshot();
    });
  });

  describe('get', () => {
    it('should call render with the "get" template', () => {
      const renderMock = jest.fn();
      const resMock = { render: renderMock };
      search.get(null, resMock);
      expect(renderMock).toHaveBeenCalledWith(search.template.get);
    });
  });
});
