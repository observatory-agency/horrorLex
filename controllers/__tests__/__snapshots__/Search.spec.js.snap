// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchController constructor should set its templates 1`] = `
Object {
  "get": "search.hbs",
}
`;
